src/main/java/cs361/battleships/models/Ship.java

```diff
@@ -50,31 +69,80 @@ public Ship(Ship other) {
 	public List<Square> getOccupiedSquares() {
 		return occupiedSquares;
 	}
-
-	public int getSize() { return size; }
+public int getSize() { return size; }
 
 	public int getCaptainHealth() { return captainHealth; }
 
-	public void place(char col, int row, boolean isVertical) {
-		for (int i=0; i<size; i++) {
+
+	public void place(char col, int row, boolean isVertical, boolean isSub) {
+
+		var sub = size;
+		if(kind == "SUBMARINE"){
+			sub -=1;
+
+		}
+
+		for (int i=0; i<sub; i++) {
+
 			if (isVertical) {
 				occupiedSquares.add(new Square(row+i, col));
+				if(isSub){
+
+					occupiedSquares.get(i).setIsSubmerged();
+				}
 				if(i == size -2){
 					occupiedSquares.get(i).setCaptain();
+
 				}
 			} else {
+
 				occupiedSquares.add(new Square(row, (char) (col + i)));
+				if(isSub){
+
+					occupiedSquares.get(i).setIsSubmerged();
+				}
 				if(i == size -2){
 					occupiedSquares.get(i).setCaptain();
+
 				}
 			}
+
 		}
+		if (kind == "SUBMARINE" && isVertical) {
+
+			occupiedSquares.add(new Square(row + 1, (char) (col - 1)));
+			if(isSub){
+
+				occupiedSquares.get(sub).setIsSubmerged();
+			}
+		} else if(kind == "SUBMARINE" && !isVertical){
+
+			occupiedSquares.add(new Square(row - 1, (char) (col + 2)));
+			if(isSub){
+
+				occupiedSquares.get(sub).setIsSubmerged();
+			}
+		}
+
 	}
 
 	public boolean overlaps(Ship other) {
+
+		if(other.getOccupiedSquares().get(0).getIsSubmerged()){
+			return false;
+		}
 		Set<Square> thisSquares = Set.copyOf(getOccupiedSquares());
+
+
 		Set<Square> otherSquares = Set.copyOf(other.getOccupiedSquares());
 		Sets.SetView<Square> intersection = Sets.intersection(thisSquares, otherSquares);
+		var count = 0;
+		for(Iterator<Square> iterator = intersection.iterator(); iterator.hasNext(); ) {
+			if(iterator.next().getIsSubmerged())
+				count += 1;
+		}
+		if(count == intersection.size())
+			return false;
```

Comment:
Since all squares are submerged for any submerged ship, wouldn't it be simpler to just check if either ship is submerged then use the method from before?